cli/command/deploy.go

```diff
@@ -1,15 +1,169 @@
 package command
 
-import "github.com/urfave/cli"
+import (
+	"fmt"
+	"github.com/quintilesims/layer0/common/models"
+	"github.com/urfave/cli"
+	"io/ioutil"
+	"strconv"
+)
 
 type DeployCommand struct {
 	*CommandBase
 }
 
 func NewDeployCommand(b *CommandBase) *DeployCommand {
-	return &DeployCommand{b}
+	return &DeployCommand{
+		CommandBase: b,
+	}
 }
 
 func (d *DeployCommand) Command() cli.Command {
-	return cli.Command{}
+	return cli.Command{
+		Name:  "deploy",
+		Usage: "manage layer0 deploys",
+		Subcommands: []cli.Command{
+			{
+				Name:      "create",
+				Usage:     "create a new deploy",
+				Action:    d.create,
+				ArgsUsage: "PATH NAME",
+			},
+			{
+				Name:      "delete",
+				Usage:     "delete a deploy",
+				ArgsUsage: "NAME",
+				Action:    d.delete,
+			},
+			{
+				Name:      "get",
+				Usage:     "describe a deploy",
+				Action:    d.read,
+				ArgsUsage: "NAME",
+			},
+			{
+				Name:      "list",
+				Usage:     "list all deploys (only the latest versions of each family will be shown)",
+				Action:    d.list,
+				ArgsUsage: " ",
+				Flags: []cli.Flag{
+					cli.BoolFlag{
+						Name:  "all",
+						Usage: "list all versions of all deploys",
+					},
+				},
+			},
+		},
+	}
+}
+
+func (d *DeployCommand) create(c *cli.Context) error {
+	args, err := extractArgs(c.Args(), "PATH", "NAME")
+	if err != nil {
+		return err
+	}
+
+	content, err := ioutil.ReadFile(args["PATH"])
+	if err != nil {
+		return err
+	}
+
+	req := models.CreateDeployRequest{
+		DeployName: args["NAME"],
+		DeployFile: content,
+	}
+
+	jobID, err := d.client.CreateDeploy(req)
+	if err != nil {
+		return err
+	}
+
+	return d.waitOnJobHelper(c, jobID, "creating", func(deployID string) error {
```

Comment:
[nit] I'm not sure how everyone else feels about this, but I've generally been [a little more verbose with the spinner text](https://github.com/quintilesims/layer0/blob/cli-refactor-loadbalancer-command/cli/command/load_balancer.go#L182). I'm curious about your thoughts about being more explicit about what the spinner shows.

Reply:
We should have a consistent text style for all these `waitOnJobHelper()` calls --- most of them appear to be "creating" at present moment (or, at least, the ones I checked). I think we should decide on this outside of this PR discussion, though.

Reply:
:white_check_mark: